flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/TieredStorageMemoryManagerImpl.java

```diff
@@ -335,19 +338,21 @@ private MemorySegment requestBufferBlockingFromPool() {
     }
 
     /** @return a memory segment from the internal buffer queue. */
-    private MemorySegment requestBufferBlockingFromQueue() {
+    private MemorySegment requestBufferFromQueue() {
         CompletableFuture<Void> requestBufferFuture = new CompletableFuture<>();
         scheduleCheckRequestBufferFuture(
                 requestBufferFuture, INITIAL_REQUEST_BUFFER_TIMEOUT_FOR_RECLAIMING_MS);
 
+        hardBackpressureTimerGauge.markStart();
         MemorySegment memorySegment = null;
         try {
-            memorySegment = bufferQueue.take();
+            memorySegment = bufferQueue.poll(100, TimeUnit.MILLISECONDS);
```

Comment:
If the segment is not polled in 100ms and a null may be returned, `requestBufferBlocking` may throw a NPE because of a null segment, is it expected?

Reply:
The `requestBufferBlocking` calls this method in a `while (memorySegment == null)` loop, so it would not throw NPE. But I indeed need to add a `@Nullable` annotation to this method.